src/main/scala/sigmastate/types.scala

```diff
@@ -332,6 +353,34 @@ case class SMethod(
     this.copy(docInfo = docInfo.map(i => i.copy(isOpcode = value)))
 }
 
+object CheckTypeWithMethods extends ValidationRule(1011,
```

Comment:
Move to `ValidationRules.scala`

Reply:
Because the rules are single purpose, I would keep the rules close to the single place where they are used.